corehq/ex-submodules/casexml/apps/case/xml/parser.py

```diff
@@ -171,6 +171,23 @@ def __init__(self, identifier, attachment_src, attachment_from, attachment_name)
         self.attachment_from = attachment_from
         self.attachment_name = attachment_name
 
+    @property
+    def is_delete(self):
+        """
+        Helper method to see if this is a delete vs. update
+
+        This property is named differently from
+        `casexml.apps.case.sharedmodels.CommCareCaseAttachment.is_present`
+        to disambiguate the interface of that class from this one since
+        they are not the same but are used in very similar contexts.
+        """
+        # FIXME slight disparity between this and CaseAttachmentAction.from_v2
+        # comment: "this is a deletion if all three ... are null"
```

Comment:
The comment you're referring to doesn't include `identifier`. If there is no `identifier` then we wouldn't know which attachment to delete.

Reply:
Yes, `identifier` makes sense, it's the fact that this only checks for null `attachment_src` and `attachment_from` while the other comment references `attachment_name` as well. Other places doing deleted checks are also inconsistent with both this code and that comment. It's hard to follow because there is not one consistent way to check for deleted status. It's not even clear if that's what's being checked in some places, one has to guess. However, this is old code and I'm hesitant to change it now because that could have unintended consequences—who knows if there is code elsewhere that has come to depend on these inconsistencies? Hopefully not.

Reply:
I'd be very pro standardizing this to `delete = identifier and not (attachment_src or attachment_from)`

Reply:
I'm curious why `identifier` has to be considered for this check? I understand why we need it to actually *do* the deletion. If `src` and `from` are blank then it's a deletion; not having an identifier means we don't know what to delete. That would be a data issue, not an indication of whether it was a deletion or not.

Reply:
True, we could validate that all attachments have an identifier elsewhere though since the identifier comes from the XML element name I don't think it's possible for it to be blank at this point.

Reply:
https://github.com/dimagi/commcare-hq/pull/21106/commits/8f61e0a46f14a664a4cfb9381784e87e690ed0ec